tests/testthat/test-estimate_truncation.R

```diff
@@ -37,6 +37,36 @@ test_that("estimate_truncation can return values from simulated data with the
   expect_error(plot(est), NA)
 })
 
+test_that("estimate_truncation works with filter_leading_zeros set", {
+  skip_on_os("windows")
+  # fit model to a modified version of example_data with zero leading cases
+  # but with filter_leading_zeros = TRUE
+  modified_data <- example_truncated
+  modified_data <- purrr::map(modified_data, function(x) x[1:3, confirm := 0])
+  out <- estimate_truncation(modified_data,
+                             verbose = FALSE, chains = 2, iter = 1000, warmup = 250,
+                             stan = stan_opts(backend = "cmdstanr"),
+                             filter_leading_zeros = TRUE
+  )
+  expect_named(out, c("dist", "obs", "last_obs", "cmf", "data", "fit"))
+  expect_s3_class(out$dist, "dist_spec")
+})
+
+test_that("estimate_truncation works with zero_threshold set", {
+  skip_on_os("windows")
+  # fit model to a modified version of example_data with zero leading cases
+  # but with filter_leading_zeros = TRUE
+  modified_data <- example_truncated
+  modified_data <- purrr::map(modified_data, function(x) x[sample(1:10, 6), confirm := 0])
```

Comment:
I think this line doesn't currently do anything as all these lines are 0 anyway. Once https://github.com/epiforecasts/EpiNow2/pull/628 is merged we should merge `main` here and then run the tests again (I still think there will be an issue but I might be wrong).

Reply:
It is not a great test. I will do better checks on the object.

Reply:
I've modified the test to compare the fit of the modified datasets to that of the original dataset here in c11f3e56e8051e13fe317d5608a01faf8ce70c24.

Reply:
Great idea!